src/cpp/preprocessor.c

```diff
@@ -150,15 +172,18 @@ static void handle_include(const char *p, Stream *stream) {
   char *path = strndup(p, q - p);
   char *fn = NULL;
   FILE *fp = NULL;
+  char *dir = dirname(strdup(stream->filename));
```

Comment:
I encountered test failed when I run the test on M1 Mac (which uses standard libc library on xcc).

It says `dirname()` function might return inner static buffer,
and it is overridden in second call in `fullpath()` called from `search_sysinc_next()`.

It looks wired, but please duplicate `direname`'s result:

```c
  char *dir = strdup(dirname(strdup(stream->filename)));
```